src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4489,7 +4489,7 @@ jint os::init_2(void) {
   // Need to ensure we've determined the process's initial stack to
   // perform the workaround
   Linux::capture_initial_stack(JavaThread::stack_size_at_create());
-  workaround_expand_exec_shield_cs_limit();
+  Linux::workaround_expand_exec_shield_cs_limit();
```

Comment:
You could move `workaround_expand_exec_shield_cs_limit()` into this file, declare it static, and remove all mentionings from headers. It now lives in os_linux_x86.cpp, but since it has to be wrapped with !zero !64bit already it is an odd misfit anyway. So its not much more "wrong" for it to live here.

Reply:
Fixed.